File: composeApp/src/commonMain/kotlin/com/crosspaste/realm/sync/SyncState.kt
package com.crosspaste.realm.sync

class SyncState {

    companion object {
        const val CONNECTED = 0
        const val CONNECTING = 1
        const val DISCONNECTED = 2
        const val UNMATCHED = 3
        const val UNVERIFIED = 4
        const val INCOMPATIBLE = 5
    }
}

File: web/src/shared/sync/sync-state.ts
/**
 * Device sync states, numerically aligned with the desktop
 * `com.crosspaste.db.sync.SyncState` constants. Do NOT reorder.
 */
export enum SyncState {
  CONNECTED = 0,
  CONNECTING = 1,
  DISCONNECTED = 2,
  UNMATCHED = 3,
  UNVERIFIED = 4,
  INCOMPATIBLE = 5,
}

export function isReachable(state: SyncState): boolean {
  return state === SyncState.CONNECTED;
}

export function isErrorState(state: SyncState): boolean {
  return (
    state === SyncState.DISCONNECTED ||
    state === SyncState.UNMATCHED ||
    state === SyncState.INCOMPATIBLE
  );
}
